components/console/single_command_tool.rst

```diff
@@ -6,68 +6,52 @@ Building a single Command Application
-            return 'my_command';
-        }
-
-        /**
-         * Gets the default commands that should always be available.
-         *
-         * @return array An array of default Command instances
-         */
-        protected function getDefaultCommands()
-        {
-            // Keep the core default commands to have the HelpCommand
-            // which is used when using the --help option
-            $defaultCommands = parent::getDefaultCommands();
+it is possible to remove this need declaring a single command application::
+
+  #!/usr/bin/env php
+  <?php
+  require __DIR__.'/vendor/autoload.php';
+
+  use Symfony\Component\Console\Application;
+  use Symfony\Component\Console\Input\InputArgument;
+  use Symfony\Component\Console\Input\InputInterface;
+  use Symfony\Component\Console\Input\InputOption;
+  use Symfony\Component\Console\Output\OutputInterface;
+
+  (new Application('echo', '1.0.0'))
+    ->register('echo')
+        ->addArgument('foo', InputArgument::OPTIONAL, 'The directory', 'foo')
+        ->addOption('bar', null, InputOption::VALUE_REQUIRED, 'Foobar', 'bar')
+        ->setCode(function(InputInterface $input, OutputInterface $output) {
+            $output->writeln('start');
+            $output->writeln($input->getArgument('foo'));
+            $output->writeln($input->getOption('bar'));
+        })
+    ->getApplication()
+    ->setDefaultCommand('echo', true)
+    ->run();
+    
+When calling your console script, the command ``echo`` will then always
+be used, without having to pass its name.
 
-            $defaultCommands[] = new MyCommand();
+Of course, you can still register a command as usual::
 
-            return $defaultCommands;
-        }
+  #!/usr/bin/env php
+  <?php
+  require __DIR__.'/vendor/autoload.php';
 
-        /**
-         * Overridden so that the application doesn't expect the command
-         * name to be the first argument.
-         */
-        public function getDefinition()
-        {
-            $inputDefinition = parent::getDefinition();
-            // clear out the normal first argument, which is the command name
-            $inputDefinition->setArguments();
+  use Symfony\Component\Console\Application;
 
-            return $inputDefinition;
-        }
-    }
+  use Acme\Command\DefaultCommand;
 
-When calling your console script, the command ``MyCommand`` will then always
-be used, without having to pass its name.
+  $application = new Application('echo', '1.0.0');
+  $command = new DefaultCommand();
 
-You can also simplify how you execute the application::
+  $application->add($command);
 
-    #!/usr/bin/env php
-    <?php
-    // command.php
+  $application->setDefaultCommand($command->getName(), true); 
+  $application->run();
 
-    use Acme\Tool\MyApplication;
+.. tip::
 
-    $application = new MyApplication();
-    $application->run();
+    You don't need to pass the command name in order to get the help.
```

Comment:
IMHO this tip is not useful, it is said [above](https://github.com/symfony/symfony-docs/pull/6876/files#diff-789a40ecb1ee9da7c5843b14bdff2ee7R8) that this is what the whole feature intends to solve.
More, it could be a warning:  if passed, the command name would be considered as the first argument of the default command, leading to whether a wrong value for this argument or an exception if no argument is expected.
> Too many arguments, expected arguments "foo"